bundles/io/org.eclipse.smarthome.io.transport.mqtt.test/src/test/java/org/eclipse/smarthome/io/transport/mqtt/test/MqttBrokerConnectionTests.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.transport.mqtt.test;
+
+import static org.junit.Assert.*;
+
+import java.util.Arrays;
+
+import javax.naming.ConfigurationException;
+
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection;
+import org.eclipse.smarthome.io.transport.mqtt.MqttConnectionObserver;
+import org.eclipse.smarthome.io.transport.mqtt.MqttMessageSubscriber;
+import org.eclipse.smarthome.io.transport.mqtt.MqttWillAndTestament;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.MockitoAnnotations;
+
+/**
+ * Tests the MqttBrokerConnection class
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class MqttBrokerConnectionTests {
+    boolean isConnected = true;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testConstructor() {
+        // Test tcp and ssl URLs
+        try {
+            MqttBrokerConnection a = new MqttBrokerConnection("name", "tcp://123.123.123.123", false);
+            MqttBrokerConnection b = new MqttBrokerConnection("name", "ssl://123.123.123.123", true);
+            assertFalse(a.isTextualConfiguredBroker());
+            assertTrue(b.isTextualConfiguredBroker());
+        } catch (ConfigurationException e) {
+            fail("Unexpected configuration exception: " + e.getMessage());
+        }
+        // Test an invalid URL
+        try {
+            @SuppressWarnings("unused")
+            MqttBrokerConnection ignored = new MqttBrokerConnection("name", "unsupported://123.123.123.123", false);
```

Comment:
I guess this is a matter of taste: I would prefer separate tests for these assertions. You may then shorten the test to something like

```
@Test(expected = ConfigurationException.class)
public void unsupportedProtocolShouldThrowException() throws Exception {
    new MqttBrokerConnection("name", "unsupported://123.123.123.123", false);
}
```